src/main/java/bio/terra/service/filedata/google/firestore/FireStoreUtils.java

```diff
@@ -29,14 +32,38 @@
             Thread.currentThread().interrupt();
             throw new FileSystemExecutionException(op + " - execution interrupted", ex);
         } catch (ExecutionException ex) {
-            // If this exception has a cause that is a runtime exception, then we rethrow the cause.
-            // Typically, the cause is one of our file system messages we want to see.
-            // If there is no cause or it is not runtime, we wrap the exception in a generic one of ours.
+            // The ExecutionException wraps the underlying exception caught in the Future, so we need
+            // to examine the properties of the cause to understand what to do.
+            // Possible outcomes:
+            // - throw FileSystemRetryException for retryable firestore exceptions to ask the step to retry
+            // - throw FileSystemExecutionException for other firestore exceptions (not retryable)
+            // - rethrow RuntimeExceptions to expose other unexpected exceptions
+            // - throw FileSystemExecutionException to wrap non-Runtime (oddball) exceptions
+
             Throwable throwable = ex.getCause();
+            while (throwable instanceof ExecutionException) {
+                ExecutionException eex = (ExecutionException)throwable;
+                throwable = eex.getCause();
+            }
+            if (throwable instanceof AbortedException) {
+                AbortedException aex = (AbortedException)throwable;
+                // TODO: in general, log + rethrow is bad form. For now, I want to make sure we see these in
+                //  the log as they happen. Once we are comfortable that retry is working properly, we can
+                //  rely on the Stairway debug logging as needed.
+                String msg = "Retrying aborted exception: " + aex.toString();
```

Comment:
`.toString()` unnecessary when using the string concatenation operator `+`. Explicitly calling `toString()` doesn't hurt except in the case where the object is null, in which case it will throw, which isn't usually what you want.